t/t0024-content-s3.t

```diff
@@ -139,6 +163,17 @@ test_expect_success HAVE_JQ 'kvs-checkpoint.get foo returns same value' '
         test_cmp value2.exp value2.out
 '
 
+test_expect_success 'changing config file does not kill running backing store' '
+    err=0 && 
+    export tmp=$FLUX_CONF_DIR && export S3_CONF_DIR=/dev/null && \
+    flux config reload && \
+	for size in $SIZES; do \
+		if ! check_blob $size; then err=$(($err+1)); fi; \
+	done && \
+    export FLUX_CONF_DIR=$tmp && \
+	test $err -eq 0
+'
+
```

Comment:
It may be nice to add a few tests for config failures here.  They should be pretty easy to check since they will just cause the `flux config reload` command to fail, but not the module.

Here is a set of tests I was playing with to verify your error handling code (which all checked out fine except the one case with `!` noted above), for your consideration:
```sh
##
# Config file reload tests
##

test_expect_success 'config: save good config' '
        mv content-s3.toml content-s3.save
'

test_expect_success 'config: reload with extra content-s3 key fails' '
        cp content-s3.save content-s3.toml &&
        echo "extrakey = 42" >>content-s3.toml &&
        test_must_fail flux config reload
'

test_expect_success 'config: reload with malformed uri fails' '
        cp content-s3.save content-s3.toml &&
        sed -i -e "s/uri =.*$/uri = \"baduri\"/" \
                content-s3.toml &&
        test_must_fail flux config reload
'

test_expect_success 'config: reload with bad credential file path fails' '
        cp content-s3.save content-s3.toml &&
        sed -i -e "s/credential-file =.*$/credential-file = \"noexist\"/" \
                content-s3.toml &&
        test_must_fail flux config reload
'

test_expect_success 'config: restore good config' '
        mv -f content-s3.save content-s3.toml &&
        flux config reload
'
```